runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/EngineBoss.java

```diff
@@ -0,0 +1,181 @@
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import org.agrona.ErrorHandler;
+import org.agrona.concurrent.Agent;
+import org.agrona.concurrent.AgentRunner;
+import org.agrona.concurrent.AgentTerminationException;
+import org.agrona.concurrent.BackoffIdleStrategy;
+import org.agrona.concurrent.IdleStrategy;
+
+import io.aklivity.zilla.runtime.engine.EngineConfiguration;
+import io.aklivity.zilla.runtime.engine.EngineController;
+import io.aklivity.zilla.runtime.engine.binding.Binding;
+import io.aklivity.zilla.runtime.engine.binding.BindingController;
+import io.aklivity.zilla.runtime.engine.config.BindingConfig;
+import io.aklivity.zilla.runtime.engine.config.KindConfig;
+import io.aklivity.zilla.runtime.engine.config.NamespaceConfig;
+import io.aklivity.zilla.runtime.engine.internal.poller.Poller;
+import io.aklivity.zilla.runtime.engine.poller.PollerKey;
+
+public class EngineBoss implements EngineController, Agent
+{
+    private static final String AGENT_NAME = "EngineBoss";
+
+    private final AgentRunner runner;
+
+    private final Poller poller;
+    private final Deque<Runnable> taskQueue;
+    private final Map<String, BindingController> controllersByType;
+
+    private volatile Thread thread;
+
+    public EngineBoss(
+        EngineConfiguration config,
+        ErrorHandler errorHandler,
+        Collection<Binding> bindings)
+    {
+        this.poller = new Poller();
+        this.taskQueue = new ConcurrentLinkedDeque<>();
+
+        this.controllersByType = bindings.stream()
+        .flatMap(b ->
+        {
+            BindingController controller = b.supply(this);
+            return controller != null ? Stream.of(Map.entry(b.name(), controller)) : Stream.empty();
+        })
+        .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        IdleStrategy idleStrategy = new BackoffIdleStrategy(
+            config.maxSpins(),
+            config.maxYields(),
+            config.minParkNanos(),
+            config.maxParkNanos());
+
+        this.runner = new AgentRunner(idleStrategy, errorHandler, null, this);
+    }
+
+    @Override
+    public int doWork() throws Exception
+    {
+        int workDone = 0;
+
+        try
+        {
+            workDone += poller.doWork();
+
+            if (!taskQueue.isEmpty())
+            {
+                for (Runnable task = taskQueue.poll(); task != null; task = taskQueue.poll())
+                {
+                    task.run();
+                    workDone++;
+                }
+            }
+        }
+        catch (Throwable ex)
+        {
+            throw new AgentTerminationException(ex);
+        }
+
+        return workDone;
+    }
+
+    public void doStart()
+    {
+        thread = startOnThread(runner, Thread::new);
+    }
+
+    public void doClose()
+    {
+        try
+        {
+            controllersByType.values().forEach(BindingController::detachAll);
```

Comment:
This `detachAll()` for `EngineBoss` and `BindingController` seems inconsistent with the design of separation between `EngineWorker` and `BindingHandler`.

Instead, we can keep track of the `NamespaceConfig`s that we have attached to the `EngineBoss` and then iterate each `NamespaceConfig` to here in `doClose` and `detach` them all.